src/cdk/tree/tree.ts

```diff
@@ -100,9 +100,29 @@ export class CdkTree<T, K = T> implements AfterContentChecked, CollectionViewer,
   }
   private _dataSource: DataSource<T> | Observable<T[]> | T[];
 
-  /** The tree controller */
+  /**
+   * The tree controller
+   *
+   * @deprecated Use one of levelAccessor or childrenAccessor
+   */
   @Input() treeControl: TreeControl<T, K>;
 
+  /**
+   * Given a data node, determines what tree level the node is at.
+   *
+   * One of levelAccessor or childrenAccessor must be specified, not both.
+   * This is enforced at run-time.
+   */
+  @Input() levelAccessor: ((dataNode: T) => number)|null;
```

Comment:
Rather than making `levelAccessor` and `childrenAccessor` nullable (`|null`), which it be more accurate to make them optional (`levelAccess?: ...`)?

Reply:
I thought for some reason they were `null` if not provided. `undefined` is more accurate here, yes.